### PR TITLE
Bioc coding practice notes

### PR DESCRIPTION
**Pull request by jmitchell81**
Addresses notes associated with coding practices called by BiocCheck including:
- replacing use of sapply with vapply or lapply
- replace use of 1:n with seq or seq_along
- replace use of = for assignment with <-
- remove redundant use of paste functions in condition signals (message, warning, error)

Other fixes include replacing uses of @ in function examples with accessor function calls and specifying the logical return value of check_args

### CODE REVIEW

File: R/import_fxns.R [hunk 1]
```diff
@@ -325,7 +325,7 @@ create_domino <- function(
   if ("database_name" %in% colnames(rl_map)) {
     dom@db_info <- rl_map
     if (verbose) {
-      message(paste0("Database provided from source: ", unique(rl_map[["database_name"]])))
```
Comment (dimalvovs) [inline R/import_fxns.R:328]: I like this!

File: R/import_fxns.R [hunk 2]
```diff
@@ -366,7 +370,7 @@ create_domino <- function(
   dom@linkages$complexes <- NULL
   if (use_complexes) {
     complex_list <- list()
-    for (i in 1:nrow(rl_reading)) {
+    for (i in seq(nrow(rl_reading))) {
```
Comment (dimalvovs) [inline R/import_fxns.R:373]: are we ok that `length(seq(0))` is 2?

File: R/utils.R [hunk 3]
```diff
@@ -97,10 +97,10 @@ dom_tf_activation <- function(dom) {
 #' 
 dom_correlations <- function(dom, type = "rl") {
     if (type == "complex") {
-        corrs = slot(dom, "cor")
+        corrs <- slot(dom, "cor")
     } else if (type == "rl") {
-        misc = slot(dom, "misc")
-        corrs = misc$rec_cor
+        misc <- slot(dom, "misc")
```
Comment (dimalvovs) [inline R/utils.R:102]: like

File: R/import_fxns.R [hunk 4]
```diff
@@ -75,7 +75,7 @@ create_rl_map_cellphonedb <- function(
   }
   # Step through the interactions and build rl connections.
   rl_map <- NULL
-  for (i in 1:nrow(interactions)) {
+  for (i in seq(nrow(interactions))) {
```
Comment (kjkrishnan) [inline R/import_fxns.R:78]: it seems like seq_len might be safer than just seq for anything with just one argument (ie, seq_len(17) is preferred over seq(17) if the goal is 1:17, per seq documentation).

File: R/import_fxns.R (hunk 2 above)
Comment (kjkrishnan) [reply R/import_fxns.R:394]: the seq_len switch would fix this too, `length(seq_len(0))` is 0

**Review by kjkrishnan — COMMENTED** [review]
Looks good over all! Thanks for cleaning up the code.

**Review by snag-gh — COMMENTED** [review]
the changes look good. My recommendation was to use seq_len rather than seq - like Kavita also suggests. 

Unsure about the handling of the Uniprot to geneid

**Pull request merged**